samanthas_telegram_bot/api_queries/api_client.py

```diff
@@ -0,0 +1,371 @@
+
+        personal_info_id = await cls._create_personal_info_get_id(context)
+        user_data = context.user_data  # type: ignore[attr-defined]
+
+        result = await cls._send_to_backend(
+            context=context,
+            method=HttpMethod.POST,
+            url=API_URL_TEACHERS_LIST_CREATE,
+            data=user_data.teacher_as_dict(update=update, personal_info_id=personal_info_id),
+            expected_status_code=httpx.codes.CREATED,
+            logger=logger,
+            success_message=cls._generate_success_message_for_person_creation(
+                context=context, personal_info_id=personal_info_id
+            ),
+            failure_message=f"Failed to create adult teacher with ID {personal_info_id}",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.SUCCESS_AND_FAILURE,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+        return result is not None
+
+    @classmethod
+    async def create_teacher_under_18(
+        cls, update: Update, context: CUSTOM_CONTEXT_TYPES  # type: ignore[valid-type]
+    ) -> bool:
+        """Sends a POST request to create a teacher under 18 years old."""
+
+        personal_info_id = await cls._create_personal_info_get_id(context)
+        user_data = context.user_data  # type: ignore[attr-defined]
+
+        result = await cls._send_to_backend(
+            context=context,
+            method=HttpMethod.POST,
+            url=API_URL_YOUNG_TEACHERS_LIST_CREATE,
+            data=user_data.teacher_under_18_as_dict(
+                update=update, personal_info_id=personal_info_id
+            ),
+            expected_status_code=httpx.codes.CREATED,
+            logger=logger,
+            success_message=cls._generate_success_message_for_person_creation(
+                context=context, personal_info_id=personal_info_id
+            ),
+            failure_message=f"Failed to create young teacher with ID {personal_info_id}",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.SUCCESS_AND_FAILURE,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+        return result is not None
+
+    @staticmethod
+    def get_json(
+        url_infix: str,
+        name_for_logger: str | None = None,
+        params: dict[str, str] | None = None,
+    ) -> Any:
+        """Function for simple synchronous GET requests with logging."""
+        if not name_for_logger:
+            name_for_logger = url_infix.replace("_", " ")
+
+        logger.info(f"Getting {name_for_logger} from the backend...")
+
+        # synchronous requests are only run at application startup, so no exception handling needed
+        response = httpx.get(f"{API_URL_PREFIX}/{url_infix}/", params=params)
+
+        data = response.json()
+        logger.info(f"...received {len(data)} {name_for_logger}.")
```

Comment:
I think len(data) doesn't make sense. It's just a number of something. (fields if data is dict, and elements if it's list), so I advise either remove it, or log full `data` (it could affect disk usage)

Reply:
I know it's just a number :) I actually find it useful as I know approximately how many there are of each type. I definitely don't want to log all data as it would be like testing the backend views from bot. Also, what matters is that it's more than 1. So I'm fine with `len()` :)